flink-runtime/src/main/java/org/apache/flink/runtime/util/ZooKeeperUtils.java

```diff
@@ -89,6 +90,7 @@ public static CuratorFramework startCuratorFramework(Configuration configuration
 
 		boolean disableSaslClient = configuration.getBoolean(ConfigConstants.ZOOKEEPER_SASL_DISABLE,
 				ConfigConstants.DEFAULT_ZOOKEEPER_SASL_DISABLE);
+		System.setProperty(ZooKeeperSaslClient.ENABLE_CLIENT_SASL_KEY, String.valueOf(!disableSaslClient));
```

Comment:
Please move this logic to `ZooKeeperModule` class https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/security/modules/ZooKeeperModule.java#L49